overlord/configstate/config/transaction.go

```diff
@@ -159,12 +159,17 @@ func (t *Transaction) Get(snapName, key string, result interface{}) error {
 		return err
 	}
 
-	err = GetFromChange(snapName, subkeys, 0, t.changes[snapName], result)
-	if IsNoOption(err) {
-		err = getFromPristine(snapName, subkeys, 0, t.pristine[snapName], result)
+	// commit changes onto pristine configuration, so that get has a complete view of the config.
+	// this doesn't make them visible in the state until real Commit().
+	config, ok := t.pristine[snapName]
+	if !ok {
+		config = make(map[string]*json.RawMessage)
 	}
-
-	return err
+	snapChanges := t.changes[snapName]
+	for k, v := range snapChanges {
+		config[k] = commitChange(config[k], v)
```

Comment:
aren't we mutating prisitine here, that seems wrong? do we need to make a copy before?

Reply:
Yes we are, but it's "our" deep copy of config (that we made at the start of the transaction) so it's not affecting anything outside of transaction until Commit(); it simply gets "dirty". I don't think it's an issue, in fact in the existing code of Commit() we fetch it again from scratch, and re-apply changes afresh:
```
	err := t.state.Get("config", &t.pristine)
        ...
       // Iterate through the write cache and save each item.
	for instanceName, snapChanges := range t.changes {
      ...
```
If you think it's cleaner to keep t.pristine really pristine I can indeed make a temporary copy here.